[dependency] Upgrade Material UI to v.7

veronikaslc opened the pull request:
Updated dependencies:
`@mui/x-date-pickers`
`@mui/lab`
`@mui/icons-material`
`@mui/material` 
`jss` -> `tss-react`

Removed deprecated dependency:
`@mui/styles`

Manuals used:
https://mui.com/material-ui/migration/upgrade-to-v6/
https://mui.com/material-ui/migration/migrating-from-deprecated-apis
https://mui.com/material-ui/migration/upgrade-to-v7/
https://mui.com/material-ui/migration/migrating-from-jss/
https://mui.com/x/migration/migration-data-grid-v6/

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Update Accordion element

veronikaslc committed:
[misc] Fixed updating unmounted component

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Fixed grid and header alignments for Forms and Questionnaires

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Fix DOM errors of nested p components

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Fixed Grid styling for patient survey

veronikaslc committed:
[misc] Fixed missing unique key warning

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
- Proper Grid v.2 props passing via size object;
- Better alignments for Subject page elements;
- Better alignment for entities table action column to the right;
- Better center alignment for items in AdminDashboard;

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Consistent styling of dashboard views loading state

veronikaslc committed:
[misc] Fix error of unknown onAddSuggestion event handler for TextField component

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Fixed section header positioning

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Closer spacing between action buttons in Livetable

veronikaslc committed:
[misc] Added circular progress during form saving process

veronikaslc committed:
[dependency] Upgrade Material UI to v.7

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Update resolution for new @mui/icons-material

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Update ListItem element to ListItemButton where deprecated button prop was used

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Update Grid element

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Migrating from jss to tss-react

veronikaslc committed:
[misc] Removed artifact color property from Switch

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Migrating from deprecated APIs

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
using :first-of-type in places of :first-child to avoid console log error flow

veronikaslc committed:
[misc] Webpack update file extension resolution config

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Used @emotion/react CacheProvider in place of StyledEngineProvider

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/ExportButton.jsx:
```diff
@@ -34,20 +34,20 @@ import ResponsiveDialog from "../components/ResponsiveDialog";
   },
   withMultiSelect: {
-    "& > .MuiGrid-item:first-child" : {
+    "& > .MuiGrid-root:first-of-type" : {
```
Just curious why all the `first-child` had to be replaced by `first-of-type`

veronikaslc replied:
After upgrade console was flooded with tens of errors per each page:

`The pseudo class ":first-child" is potentially unsafe when doing server-side rendering. Try changing it to ":first-of-type"`

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/LiveTable.jsx:
```diff
@@ -426,7 +426,9 @@ function LiveTable(props) {
               :
-              ( <TableRow><TableCell colSpan={columns ? columns.length : 1}>Please wait...</TableCell></TableRow> )
+              ( <TableRow><TableCell colSpan={columns ? columns.length : 1}>
+                  <Typography className={classes.dashboardEntry}>Please wait...</Typography>
```
There's no `dashboardEntry` in `LiveTableStyle`

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/SubjectView.jsx:
```diff
@@ -160,7 +161,7 @@ function SubjectView(props) {
               filtersJsonString={filtersJsonString}
             />
-          : <Typography>No results</Typography>
+          : <Typography style={{ paddingLeft: theme.spacing(1)}}>No results</Typography>
```
I thought  `sx` was preferred and we can do `sx= {{pl: 1}}` (no need to pull in the `theme`). Is that not true in v7?

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/DicomQuestion.jsx:
```diff
@@ -330,13 +330,12 @@ function DicomQuestion(props) {
 };
 
-const StyledDicomQuestion = withStyles(QuestionnaireStyle)(DicomQuestion)
+const StyledDicomQuestion = withStyles(DicomQuestion, QuestionnaireStyle)
```
This doesn't seem to be needed, we're not even using the `classes` prop introduced by it. We can skip the `withSyles` transformation and export `DicomQuestion` directly.

veronikaslc replied:
Many styles in `QuestionnaireStyle` overwrite MUI native styles and could be applicable to the component at rendering.

marta- replied:
I'm not sure what you mean, can you give an example? As far as I can tell, since we're not using the `classes` prop introduced by `withStyles`, none of the styles defined in `QuestionnaireStyle` are ever applied.

veronikaslc replied:
Right, I was thinking too globally. However, following that we need to change about 30 files

veronikaslc replied:
```
AdminStatistics.jsx
HeaderSearchBar.jsx
createuserdialogue.jsx
ChromosomeQuestion.jsx
AdminStatistics.jsx
HeaderSearchBar.jsx
AutocreatedQuestion.jsx
BooleanQuestion.jsx
PhoneQuestion.jsx
AnswerInstructions.jsx
ResourceQuestion.jsx
VocabularyQuestion.jsx
TextQuestion.jsx
BooleanInput.jsx
Fields.jsx
MarkdownTextField.jsx
NewQuestionnaireDialog.jsx
ObjectInput.jsx
NumberInput.jsx
TextInput.jsx
ListInput.jsx
ResourceFilter.jsx
UserFilter.jsx
VocabularyFilter.jsx
EditButton.jsx
PrintButton.jsx``
```

marta- replied:
I would say it is a good opportunity to clean up the code and remove this unnecessary processing.

veronikaslc replied:
fixed in [6c2be9b](https://github.com/data-team-uhn/cards/pull/1944/commits/6c2be9bb1c2ac25c74a0e36c98ac8f71c8849a06)

marta- replied:
Unneeded `withStyles` transformation is still here.

veronikaslc replied:
Right, fixed here [9cebd81](https://github.com/data-team-uhn/cards/pull/1944/commits/9cebd81c7ea93f9ac006a8067bf5fa42ab6ac9e4)

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/FileQuestion.jsx:
```diff
@@ -306,7 +306,7 @@ function FileQuestion(props) {
         pageActive && <>
           { uploadInProgress && (
-            <Grid item className={classes.root}>
+            <Grid className={classes.root}>
```
`root` is not defined in `QuestionnaireStyle`. Probably leftover from many years ago. We can remove it when we switch from extensive custom styling to theme.

veronikaslc replied:
Let's remove now while we remember, small change won't hurt

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/PedigreeQuestion.jsx:
```diff
@@ -148,9 +148,10 @@ function PedigreeQuestion(props) {
           <Dialog fullScreen open={expanded}
             onClose={() => { setExpanded(false); }}
-            TransitionProps={{
-              onEntering: () => { openPedigree(); },
```
We don't need `onEntering` anymore?

veronikaslc replied:
Removed by accident

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/Questionnaire.jsx:
```diff
@@ -662,7 +660,7 @@ let QuestionnaireEntry = (props) => {
         data={entryData}
         type={type}
-        classes={classes}
+        upperClasses={classes}
```
Why this rename?

marta- replied:
Nevermind, I figured it out.

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/SubjectSelector.jsx:
```diff
@@ -180,7 +180,7 @@ function UnstyledNewSubjectDialog (props) {
             data={ allowedTypes?.length ? allowedTypes : data }
             renderTopToolbarCustomActions={() => {
-              return <Typography variant="h6" sx={{ paddingLeft: theme.spacing(2) }}>Select a type</Typography>;
+              return <Typography variant="h6" sx={(theme) => ({ paddingLeft: theme.spacing(2) })}>Select a type</Typography>;
```
Does the short version still work? `sx={{pl: 2}}`

veronikaslc replied:
Sure, it does https://mui.com/system/getting-started/the-sx-prop/

marta- commented on modules/patient-portal/src/main/frontend/src/patient-portal/LandingPage.jsx:
```diff
@@ -86,13 +86,13 @@ function LandingPage(props) {
               </Grid>
-              <Grid item>
+              <Grid>
                 <Grid container spacing={3} direction="column" justifyContent="center" alignItems="center">
```
I think we might need `alignItems="stretch"` here to achieve the original layout.

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Addressed codereview comments:
- Removed artifact styles
- Returned accidentally removed transition property
- Better styling codestyle

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Removed unused theme

veronikaslc committed:
[code-cleanup] Removed unnecessary withStyles() from components

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Fix filters rendering errors

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Used @emotion/react CacheProvider in place of StyledEngineProvider

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/FilterComponents/BooleanFilter.jsx:
```diff
@@ -39,7 +39,7 @@ const COMPARATORS = DEFAULT_COMPARATORS.slice().concat(UNARY_COMPARATORS);
  */
 const BooleanFilter = forwardRef((props, ref) => {
-  const { classes, defaultValue, onChangeInput, questionDefinition, ...rest } = props;
+  const { classes, defaultValue, defaultLabel, onChangeInput, questionDefinition, ...rest } = props;
```
Why was `defaultLabel` added here? it doesn't appear to be used.

veronikaslc replied:
If not separated from `...rest` explicitly, it gets passed down to components that do not have that property in specs, which triggers console errors and recommendations to remove it.

marta- replied:
It's not used in this component either, so it doesn't make it better semantically.

I see it comes from https://github.com/data-team-uhn/cards/blob/dev/modules/data-entry/src/main/frontend/src/dataHomepage/Filters.jsx#L316-L317 - we explicitly pass both `defaultValue` and `defaultLabel` to all filter components and let them choose which they use. I would suggest some refactoring to avoid warnings and keep the code clean: pass both {value, label} as an object to filter components and have them use either the value or the label field as needed:

Filters.jsx line 313 - 

```
<CachedComponent
        ref={focusRef}
        questionDefinition={questionDefinitions[filterDatum.name]}
        defaultValue={editingFilters[index].value}
        defaultLabel={editingFilters[index].label}
        onChangeInput={(newValue, label) => {handleChangeOutput(index, newValue, label, dataType);}}
        />);
```
->
```
<CachedComponent
        ref={focusRef}
        questionDefinition={questionDefinitions[filterDatum.name]}
        default={{
          value: editingFilters[index].value,
          label: editingFilters[index].label
        }}
        onChangeInput={(newValue, label) => {handleChangeOutput(index, newValue, label, dataType);}}
        />);
```

veronikaslc replied:
default can not be used as a variable prop name, it is a reserved keyword in JavaScript

marta- replied:
Right, my bad, same idea under a different name.

veronikaslc replied:
Would defaultFilterProps be OK?

marta- replied:
Looking at the code in Filters.jsx, `default` seems a poor name choice anyway. Those look more like `current` value and label to me. Better name options are of course welcome as always.

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/FilterComponents/ListFilter.jsx:
```diff
@@ -39,7 +39,7 @@ const COMPARATORS = DEFAULT_COMPARATORS.slice().concat(UNARY_COMPARATORS);
  */
 const ListFilter = forwardRef((props, ref) => {
-  const { classes, defaultValue, onChangeInput, questionDefinition, ...rest } = props;
+  const { classes, defaultValue, defaultLabel, onChangeInput, questionDefinition, ...rest } = props;
```
Also here

marta- replied:
And in several other filter renderers

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/Form.jsx:
```diff
@@ -84,6 +86,7 @@ function Form (props) {
   let [ error, setError ] = useState();
   // Marks that a save operation is in progress
   let [ saveInProgress, setSaveInProgress ] = useState();
+  let [ fetchInProgress, setFetchInProgress ] = useState();
```
Does the introduction of `fetchInProgress` belong in this PR? IIRC it was needed to mask some rendering lag that came with the router upgrade. If that's correct, it should me moved to that PR.

veronikaslc replied:
Removed in [3116bbd](https://github.com/data-team-uhn/cards/pull/1944/commits/3116bbd426f4c6f63f4e3e889f3f6a9a2544d125)

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/NumberQuestion.jsx:
```diff
@@ -54,13 +53,13 @@ const DATA_TO_VALUE_TYPE = {
 };
 
-const useSliderStyles = makeStyles(theme => ({
+const useSliderStyles = makeStyles()(theme => ({
```
At line 174 we still do `const sliderClasses = useSliderStyles();` instead of `const { classes } = useSliderStyles();`, which means we don't really apply anything defined by `useSliderStyles()`. A quick fix is to replace all later occurrences of `sliderClasses` with `sliderClasses.classes`. A better approach would be to refactor `NumberQuestion` and pull everything related to sliders in a separate component, but that can be done as a separate task.

veronikaslc replied:
Quick fixed in [60dc57c](https://github.com/data-team-uhn/cards/pull/1944/commits/60dc57c2297e1c0481370e55aea3a212eeda4891)

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/QuestionnaireStyle.jsx:
```diff
@@ -202,6 +202,9 @@ const questionnaireStyle = theme => ({
          "& .MuiTableCell-body": {
             padding: theme.spacing(0, 2),
          },
+         "& .MuiTableCell-body:last-child": {
+            paddingRight: theme.spacing(.5),
+         },
```
Why is this needed?

veronikaslc replied:
Without 
![image](https://github.com/user-attachments/assets/48ce952a-148c-4630-9981-bc39183293aa)
With 
![image](https://github.com/user-attachments/assets/65a1451b-2d96-412f-941f-2eb67a5f2859)

marta- commented on modules/data-entry/src/main/frontend/src/questionnaire/QuestionnaireStyle.jsx:
```diff
@@ -257,7 +260,7 @@ const questionnaireStyle = theme => ({
     },
     subjectNestedContainer: {
-        marginLeft: theme.spacing(5),
+        paddingLeft: theme.spacing(5),
```
Why padding instead of margin?

veronikaslc replied:
With margin
![image](https://github.com/user-attachments/assets/f22b22c6-d04e-4714-b944-1562c4f81237)
With padding
![image](https://github.com/user-attachments/assets/84438222-34e6-4adc-bf35-a0f4cd8c4332)

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Quick fix for NumberQuestion style classes

veronikaslc committed:
[dependency] Upgrade Material UI to v.7

Removed withStyles() from DicomQuestion component

veronikaslc committed:
[code-cleanup] Refactored and unify Filters styling

veronikaslc committed:
[misc] Removed updates related to the react router upgrade

veronikaslc commented:
Refactored Filters to make them look more consistent
Before:
![before](https://github.com/user-attachments/assets/724f8f20-abee-47e5-a3fd-9ad670aa175a)
After:
![after](https://github.com/user-attachments/assets/8739b9bc-15d5-4233-bd45-af2d7771254a)

veronikaslc commented:
Also removed HeaderSearchBar component as it it not needed.

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/FilterComponents/TextFilter.jsx:
```diff
@@ -39,16 +39,16 @@ const QuestionnaireStyleNotesContain = theme => ({
  */
 const TextFilter = forwardRef((props, ref) => {
-  const { classes, defaultValue, defaultLabel, onChangeInput, questionDefinition, ...rest } = props;
+  const { classes, current, onChangeInput, questionDefinition } = props;
```
@veronikaslc you are right in your comments, `initial` is a much better name. Let's rename please.

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/FilterComponents/TextFilter.jsx:
```diff
@@ -39,16 +39,16 @@ const QuestionnaireStyleNotesContain = theme => ({
  * getFilterComparatorsAndComponent method.
  *
- * @param {string} defaultValue The default value to place in the text field
+ * @param {object} current Object containing the initial value and label to place in the text field
```
@veronikaslc you are right in your code comments, `initial` is a much better name than `current`. Let's rename please.

marta- commented:
Notes are too small.

Dev:
![image](https://github.com/user-attachments/assets/f934939c-815f-4c65-94ce-b8432a249a94)

This branch:
![image](https://github.com/user-attachments/assets/133d116f-9a13-4d74-af20-b0333f0fbcba)

veronikaslc committed:
[code-cleanup] Refactored and unify Filters styling
Better name for initial filter props object

veronikaslc commented:
@marta- Which questionnaire were you testing for notes?

marta- commented:
> @marta- Which questionnaire were you testing for notes?

The screenshot is from `Concept Recognition Test`. Any questionnaire with any question with notes enabled will have the same issue (notes too narrow).

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Fixed Notes width

marta- commented on modules/commons/src/main/frontend/src/SearchBar.jsx:
```diff
@@ -197,7 +194,7 @@ function SearchBar(props) {
           <InputAdornment position="end">
             <IconButton
               size="large"
-              className={invertColors ? classes.invertedColors : ""}
+              className={(invertColors ? classes.invertedColors + " " : "") + classes.searchIcon}
```
I would make the button size `small` and get rid of `classes.searchIcon`.

marta- commented on modules/homepage/src/main/frontend/src/themePage/Navbars/AdminNavbarLinks.jsx:
```diff
@@ -113,10 +115,13 @@ function HeaderLinks (props) {
           invertColors={!expand}
           onSelectFinish={expand ? undefined : closeSidebar}
-          className={expand ? undefined : classes.buttonLink}
+          className={classNames(classes.search, {[classes.buttonLink]: !expand})}
```
There's no `search` defined in `sidebarStyle`, only in `headerStyle`, so moving `classes.search`  here from `SearchBar` (which is styled with `headerStyle`) does not have the desired effect.

veronikaslc replied:
There is a search in sidebarStyle, I moved the class specifically there from headerStyle in the same commit so it can have effect.

marta- replied:
Right, i see it. This PR definitely has too many changes for my attention span :)

veronikaslc committed:
[code-cleanup] Refactored and unify Filters styling
code cleanup

veronikaslc commented:
Now looks like this:
![berore](https://github.com/user-attachments/assets/26405563-4668-4ca5-a5a6-37efc1c9a1f7)

marta- commented on modules/data-entry/src/main/frontend/src/resourceQuery/ResourceQuery.jsx:
```diff
@@ -142,7 +142,7 @@ function ResourceQuery(props) {
         setSuggestionsVisible(false);
         setResourcePath("");
       }}
-      className={variant == "labeled" ? classes.searchInput : "" + className}
```
Why this change?

veronikaslc replied:
Because we don't need "" +

marta- commented:
Unless I'm missing something, `ResourceFilter`/`ResourceQuery` do not rely on the `SearchBar` component or its style, so they're not affected by the removal of `searchIcon` from `headerStyle`. These filters looks the same in `dev`.

veronikaslc commented:
ResourceFilter/ResourceQuery do not rely on the SearchBar, Subject Filter does. Now it's adornment icon is misallined related to others, I made a screenshot to show it. It looks like in dev because that how it is now in dev before my changes that we reverted.

marta- commented:
> ResourceFilter/ResourceQuery do not rely on the SearchBar, Subject Filter does. Now it's adornment icon is misallined related to others, I made the lastest screenshot above to show it. It looks like in dev because that how it is now in dev before my changes that we reverted.

My mistake, again. Let's shift the search icon to align it using `sx`.

marta- commented:
Patient portal alignment issue: text should be centered:

![image](https://github.com/user-attachments/assets/6d523cc2-b143-4ddd-92f8-9ebf5ee5870f)

veronikaslc committed:
[code-cleanup] Refactored and unify Filters styling
Added disableButton prop to the SearchBar component

sdumitriu commented on modules/commons/src/main/frontend/src/SearchBar.jsx:
```diff
@@ -50,11 +49,12 @@ const CARDS_QUERY_MATCH_PATH_KEY = "@path";
  * @param {func} resultConstructor Function that constructs a DOM element from a row of results.
  * @param {bool} showAllResultsLink If true, show the link “See all results” of the bottom of the results dropdown
  * @param {bool} disableDropdownItemLink If true, disable links for results dropdown items
+ * @param {bool} disableButton If true, puts just the search icon at the end instead o the functional search button
```
```suggestion
 * @param {bool} disableButton If true, puts just the search icon at the end instead of the functional search button
```

sdumitriu commented on modules/commons/src/main/frontend/src/SearchBar.jsx:
```diff
@@ -75,16 +75,18 @@ function SearchBar(props) {
-  let getQuickSearchSettings = () => {
-    fetchWithReLogin(globalLoginDisplay, '/apps/cards/config/QuickSearch.json')
+  useEffect(() => {
+    if (!fetched) {
```
Is this even needed? The `useEffect` ensures that it will only be used once, so it's safe to remove the `fetched` state.

veronikaslc committed:
[dependency] Upgrade Material UI to v.7

Replaced button prop deprecated for ListItem in v.5 with ListItemButton

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Followed React's convention of using camelCase for CSS properties in style objects.

veronikaslc commented on modules/homepage/src/main/frontend/src/themePage/Navbars/QuickSearchIdentifier.jsx:
```diff
@@ -83,7 +83,7 @@ function MatchAvatar(props) {
 
 function ListItemLink(props) {
-  return <ListItem alignItems="center" button component="a" {...props} />;
+  return <ListItemButton alignItems="center" component="a" {...props} />;
```
See for details https://mui.com/material-ui/migration/upgrade-to-v6/#listitem

veronikaslc committed:
[misc] Removed conflicling class from PatientIdentification

sdumitriu commented on modules/homepage/src/main/frontend/src/themePage/Navbars/AdminNavbarLinks.jsx:
Line 187, `<Snackbar` still uses the deprecated `ContentProps` instead of `slotProps.content`.

sdumitriu commented on modules/commons/src/main/frontend/src/SearchBar.jsx:
Line 253, `primaryTypographyProps` should be replaced with `slotProps.primary`.

sdumitriu commented on modules/data-entry/src/main/frontend/src/dataHomepage/VariableAutocomplete.jsx:
In a future PR, since MUI is moving from individual `somethingProps` to `slotProps`, we should also change the way we pass component props between our component to something similar, for example here we have `textFieldProps`, and in `AdminResourceListing` we have `buttonProps`.

marta- commented on modules/data-entry/src/main/frontend/src/resourceQuery/ResourceQuery.jsx:
```diff
@@ -140,7 +142,7 @@ function ResourceQuery(props) {
         setResourcePath("");
       }}
-      className={variant == "labeled" ? classes.searchInput : ""}
+      className={variant == "labeled" ? classes.searchInput : "" + className}
```
```suggestion
      className={(variant == "labeled" ? (classes.searchInput + " ")  : "") + className}
```

marta- commented on modules/commons/src/main/frontend/src/SearchBar.jsx:
```diff
@@ -195,22 +193,20 @@ function SearchBar(props) {
+              ? <Search/>
+              : <IconButton
+                  size="small"
+                  sx={{mr: 0.5}}
```
The button is getting shifted in the wrong direction, creating an even bigger gap on the right.

How it is now:
![image](https://github.com/user-attachments/assets/cc1b88f6-3d4e-40aa-94e2-90c591db2257)

How we want it:
![image](https://github.com/user-attachments/assets/16d65789-22c2-4d7e-b4ee-c48935f16738)


```suggestion
                  sx={{mr: -0.5}}
```

marta- commented:
> Patient portal alignment issue: text should be centered:

I should have been more specific. The text before and after the list should be centered. The list should remain left aligned and the Alert should remain left-aligned.

How it looks on `dev` (this is what we want):

![image](https://github.com/user-attachments/assets/12d57092-151b-4fd4-b4b2-33438bd87f66)

How it looks now on this branch:

![image](https://github.com/user-attachments/assets/e87ccd9d-61cb-43c6-b8bd-68c9690e0284)

marta- commented:
The form previews on the Review page are centered, should be left aligned.

![image](https://github.com/user-attachments/assets/dc227362-8db8-4782-93c6-7353c1d14092)

marta- commented:
The score interpretations are centered, should be left aligned:

![image](https://github.com/user-attachments/assets/6d2c0df1-a3a1-4829-82e3-a38c36da5d63)

veronikaslc commented:
Are you sure we want the text before the list to be centered? It looks bad ihmo. The usual practice of mass amount of text is to left center it
![image](https://github.com/user-attachments/assets/10dd37c4-e431-4404-9a50-b8c577b579c7)

veronikaslc commented:
I removed commit that centered all text that affected patient portal pages

veronikaslc commented:
What we have now in the PR imho looks better not centered
![image](https://github.com/user-attachments/assets/4b9aaca3-725b-488a-856a-704048a150fe)

veronikaslc commented:
I suggest that would even look better
![image](https://github.com/user-attachments/assets/da2ba069-7645-4fd8-8643-b7f4aa35ffb1)

veronikaslc commented:
Also the text at the bottom looks better if not center alligned as it is a list of notes to the user

veronikaslc committed:
[misc] Better filter item adornment icon alignment and fixed order of operations and combining classes

marta- commented on modules/patient-portal/src/main/frontend/src/patient-portal/PatientIdentification.jsx:
```diff
@@ -349,16 +346,16 @@ function PatientIdentification(props) {
               </Grid>
-              <Grid item className={classes.identifierDivider}>or</Grid>
-              <Grid item>
+              <Grid className={classes.identifierDivider}>or</Grid>
```
Since we're cleaning up the styling in Patient identification, this is better: 
```suggestion
              <Grid alignSelf="center">or</Grid>
```
(also remove the now unused `identifierDivider` class name at line 81.

veronikaslc committed:
[misc] Cleaning up the styling in Patient identification

marta- commented:
I really do not think this looks good:

![image](https://github.com/user-attachments/assets/d8111ea5-cf19-44d6-bc08-5f2e267f2880)

This ^ is exactly how patients will see it. While we can rephrase all the YE intro texts to make them look better left-aligned, now is not the the time and this PR is not the place to do it (Patient Experience may need to be involved).

Let's keep the layout as in `dev`. We can make a separate issue to improve it in the future,

acrowthe commented:
I much prefer the centered version overall. I prefer the left aligned blocks of text slightly but centered large blocks looks reasonable. The centered shorter text looks way better than shorter left aligned so that's my preference

veronikaslc commented:
We can not keep layout as dev with this version of MUI. It generates 3 places of nested <p>. Reverting [this commit](https://github.com/data-team-uhn/cards/pull/1944/commits/43150edca75f8afb51af5a519c098967bd0520e5) will bring up this issue back.

marta- committed:
[regression-fix] Restored center alignment for patient portal survey instructions

The layout had changed to left-aligned in commit 43150ed which fixed nested p elements.

marta- commented:
> We can not keep layout as dev with this version of MUI. It generates 3 places of nested `<p>`. Reverting [this commit](https://github.com/data-team-uhn/cards/pull/1944/commits/43150edca75f8afb51af5a519c098967bd0520e5) will bring up this issue back.

There's always a way :) Addressed in 1cb20fe .

veronikaslc commented:
In general as we are developing the app for the users we should prioritise the usability over our personal preferences of any points. As far as I understand the goal, it is not what we personally like or aesthetically prefer is important but what is comfortable for the end user group and brings an effective and productive clear user experience in a first place. If we focus on the most effective workflow of filling the questionnaires, the text presentation is playing the center role here. The spacing, the alignment and decorations all are the equally influetial on user-computer interaction effectiveness.

There was [a research](https://www.sciencedirect.com/science/article/abs/pii/S0141938207000133) conducted to stydy how text alignment is affecting the content disgesting and impacts how the brain processes information visually and cognitively for the production of web pages. This study highlights the importance of prioritizing readability over pure aesthetics in text alignment. The spoiler is that _left-aligned body text led to better performance, although participants preferred justified text_. Anything text-centered whch is more than 3 lines will create additional load of eye movement, brain processing and extra hardness to follow.
that expecially holds for those with reading difficulties or who may use screen readers. Centering can be a barrier for users with disabilities, as it can make it harder for them to scan and read the text. 

When centering could be an excellent tool for placing short call-for-attension elements to ephacize, like a link/action button, shourt quote, or navigate, like topic headings, it is not favourable to use centering for listing numerous user instructions or presenting introduction texts. Those are usual commom disign priciples of web presentations as I see them in current.

marta- commented:
> In general as we are developing the app for the users we should prioritise the usability over our personal preferences of any points

And that is why I suggested to create a separate task to improve the layout. This is about upgrading a dependency and should remain focused on its purpose.

veronikaslc committed:
[misc] Comment typo fix

Co-authored-by: Sergiu Dumitriu <sdumitriu@users.noreply.github.com>

veronikaslc committed:
[code-cleanup] Removed useless fetched status

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
More slotProps applications to previous v.5 props handling way

veronikaslc commented:
>  This is about upgrading a dependency and should remain focused on its purpose.
Total support :) Current comments are addressed, fells like @mdpham can grab for testing?

veronikaslc committed:
CARDS-2671: After changing the password, a confirmation alert is displayed in the wrong place

veronikaslc committed:
[misc] Fixed change password error styling

veronikaslc committed:
CARDS-2671: After changing the password, a confirmation alert is displayed in the wrong place

Better Snackbar positioning

Co-authored-by: Marta Girdea <651980+marta-@users.noreply.github.com>

veronikaslc committed:
[misc] Removed unnecessary Grid container from change user password dialog

sdumitriu committed:
Merge pull request #1958 from data-team-uhn/CARDS-2671

CARDS-2671: After changing the password, a confirmation alert is displayed in the wrong place

sdumitriu commented on modules/demo-banner/src/main/frontend/src/demoBanner.jsx:
There is a regression, in mobile mode the icon overlaps the text.
before:
![image](https://github.com/user-attachments/assets/e49dd56c-bba7-471d-96cb-24843be603ef)
after:
![image](https://github.com/user-attachments/assets/4a0eabbf-7162-4342-b262-7a69ff3323ba)

sdumitriu commented on modules/downtime-warning-banner/src/main/frontend/src/downtimeBanner.jsx:
This one too has the icon overlapping the text.
![image](https://github.com/user-attachments/assets/4b999602-648f-406f-b511-e88a67fba032)

sdumitriu commented on modules/demo-banner/src/main/frontend/src/demoBanner.jsx:
Another change is that before the text was centered, now it's left aligned. I'm OK with this change, I don't think it matters if it's centered or left-aligned. What do others think?

sdumitriu replied:
Also, the downtime banner was and is left-aligned, so having the demo banner centered is inconsistent with the downtime banner.

sdumitriu commented on modules/data-entry/src/main/frontend/src/questionnaire/QuestionnairePreview.jsx:
```diff
@@ -68,12 +68,11 @@ function QuestionnairePreview (props) {
 
-  return (<div className={data?.hideAnswerInstructions ? classes.hideAnswerInstructions : null}>
-    <Grid container {...FORM_ENTRY_CONTAINER_PROPS} >
+  return (<React.Fragment>
```
This is a regression, the `hideAnswerInstructions` classname is only considered in two places, `Form` and `QuestionnairePreview`. The effect is that while filling in an actual form still hides the instructions, the preview in the administration always shows the instructions regardless of the `hide` setting.

sdumitriu commented on modules/homepage/src/main/frontend/assets.config:
```diff
@@ -1,10 +1,10 @@
-['cards-homepage.modelOrganismsIcon']: '@mui/icons-material/Pets.js'
-['cards-homepage.variantsIcon']: '@mui/icons-material/Subtitles.js'
-['cards-homepage.adminIcon']: '@mui/icons-material/Settings.js'
+['cards-homepage.modelOrganismsIcon']: '@mui/icons-material/Pets'
+['cards-homepage.variantsIcon']: '@mui/icons-material/Subtitles'
```
These 2 aren't even used, they should be removed in a separate PR.

sdumitriu commented on modules/commons/src/main/frontend/src/SearchBar.jsx:
Regression in the way quick search results look.
Before:
![image](https://github.com/user-attachments/assets/977feb87-e533-4d20-b095-4b8cadb80319)

After:
![image](https://github.com/user-attachments/assets/95386662-dee6-448c-8c68-b25329ef5a6c)

There is too much padding between the two gray backgrounds, wasting space.

marta- replied:
This doesn't appear to have been addressed yet @veronikaslc

veronikaslc replied:
Already addressed: extra padding removed, input width improved from dev to align with dropdown
Dev
![dev](https://github.com/user-attachments/assets/5e4650b5-008f-401c-8f45-53053fc5b320)

Mui
![mui](https://github.com/user-attachments/assets/5c798a64-f7bd-48d5-a39f-1a2fed0a55cc)

marta- replied:
Is it possible to make it so there aren't 2 shades of gray, just one? No big deal now if too hard.

sdumitriu replied:
It's more complicated, I say we fix it later.

sdumitriu replied:
A quick fix is to switch the padding from the outer element to the inner one, but then we need more style to also add padding for the "searching...", "No results" and "See all results" items which don't have an inner element. Or also add the inner element to those things. But like I said, this is more changes than needed.

sdumitriu commented on modules/homepage/src/main/frontend/src/themePage/Navbars/AdminNavbarLinks.jsx:
The position of the avatar changed slightly, and I like the old horizontal position more, but the new vertical position.
Before:
![image](https://github.com/user-attachments/assets/1cdf242a-9d19-4314-9dbf-ca0cef9e20b7)

After:
![image](https://github.com/user-attachments/assets/6cf78c73-8795-4b84-90dd-59184e5367c5)

Can you try to make sure it's aligned with the right end of the Forms box?

sdumitriu replied:
In small mode, this also affects the hamburger icon:
![image](https://github.com/user-attachments/assets/bdbab17d-6a7a-4d8b-97e2-8847685750e2)

sdumitriu commented on modules/patient-portal/src/main/frontend/src/patient-portal/QuestionnaireSet.jsx:
```diff
@@ -708,30 +710,28 @@ function QuestionnaireSet(props) {
     </Grid>
   ] : [
     <Typography variant="h4" key="review-title">Review and Submit</Typography>,
-    <Divider/>,
```
Why remove the dividers?

veronikaslc replied:
With divider
![withdivider](https://github.com/user-attachments/assets/2b590d67-74f4-4bda-aa8c-f7e2341e9cae)

Without
![withoutdivider](https://github.com/user-attachments/assets/2e2461c8-3f47-47e6-a8a6-ff1339a7f214)

veronikaslc replied:
Too much space was occupying top of the screen between the header and first submit button.

sdumitriu reviewed (changes requested):
I get this error when first displaying some of the tables:
> There was a problem inserting the following rule: ".tss-gur5lp-MuiTablePagination-root . MuiTablePagination-select{margin:0 1px;}" DOMException: An invalid or illegal string was specified

I think it's not from our code, but a bug in MuiTable? To trigger it, start adding a new Questionnaire from the dashboard, it shows as soon as the list of questionnaires is loaded, or go to Administration -> Questionnaires, Users, Clinics or other pages with a table.

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Align top right naviagtion/avatar icon to the right top side;
Removed unused styling class

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Align quick result search with the search input width, remove extra padding

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Revert changes for QuestionnairePreview Grid

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Fixed demo banner and downtime warning icon grid min width

sdumitriu commented on modules/homepage/src/main/frontend/src/themePage/Sidebar/sidebarStyle.jsx:
```diff
@@ -379,7 +379,7 @@ import {
     },
     logout: {
-      paddingRight: theme.spacing(2)
+      paddingRight: theme.spacing(0.5)
```
This changed the shape of the aura in a not nice way:
![image](https://github.com/user-attachments/assets/d3915e30-59c1-4c34-b9f8-fc8dca175b84)

sdumitriu committed:
[dependency] Upgrade Material UI to v.7
Better alignment for the avatar

sdumitriu committed:
[dependency] Upgrade Material UI to v.7
Better alignment for the drawer toggle button

marta- commented:
![image](https://github.com/user-attachments/assets/c649575d-b53e-4487-9d8b-475105950217)

sdumitriu committed:
[dependency] Upgrade Material UI to v.7
Improve alignment of the navbar and the content

veronikaslc committed:
[dependency] Upgrade Material UI to v.7
Removed required classes prop from DateQuestionYear propTypes
